src/atoms/affine/index.jl

```diff
@@ -78,12 +78,12 @@ end
 
 ## API Definition begins
 
-getindex{T <: Real}(x::AbstractExpr, rows::AbstractArray{T, 1}, cols::AbstractArray{T, 1}) = IndexAtom(x, rows, cols)
-getindex{T <: Real}(x::AbstractExpr, inds::AbstractArray{T, 1}) = IndexAtom(x, inds)
+getindex(x::AbstractExpr, rows::AbstractArray{<:Real, 1}, cols::AbstractArray{<:Real, 1}) = IndexAtom(x, rows, cols)
```

Comment:
In this case the previous method was ensuring that both `T`s were the same. Was that not necessary?

Reply:
I am not sure it's necessary, but I'll change it back to the same type.